neml/nlsolvers.py

```diff
@@ -15,6 +15,54 @@ class MaximumSubdivisions(RuntimeError):
   def __init__(self):
     super(MaximumSubdivisions, self).__init__("Exceeded the maximum allowed step subdivisions!")
 
+def newton_precalc(RJ, x0, verbose = False, rtol = 1.0e-6, atol = 1.0e-10, 
+    miter = 50):
+  """
+    Manually-coded newton-raphson with precalculated jacobian
+
+    This version doesn't do linesearch because ironically that would 
+    require re-inverting everything.
+
+    Parameters:
+      RJ            function return the residual + a function that gives
+                    the action of the inverse jacobian on a vector
+      x0            initial guess
+
+    Optional:
+      verbose       verbose output
+      rtol          relative tolerance
+      atol          absolute tolerance
+      miter         maximum iterations
+  """
+  R, J = RJ(x0)
+  nR = la.norm(R)
+  nR0 = nR
+  x = np.copy(x0)
```

Comment:
This is the only comment I have and you can decide on your discretion.
I always prefer to avoid code repetition, however I'm not sure about efficiency.
It would be nice to mimic a do while loop, but this would add a condition on the firs iteration to update nR0. beside this the code is ok

Reply:
I did that for efficiency.  For now let's just leave it.